msu/systems/serialization/serialization_types/array_data.nut

```diff
@@ -0,0 +1,55 @@
+::MSU.Serialization.Class.ArrayData <- class extends ::MSU.Serialization.Class.AbstractData
+{
+	__DataArray = null;
+
+	constructor( _data )
+	{
+		::MSU.requireArray(_data);
+
+		base.constructor(::MSU.Serialization.DataType.Array, _data);
+
+		this.__DataArray = array(_data.len());
+		foreach (i, value in _data)
+		{
+			this.__DataArray[i] = ::MSU.Serialization.__convertValueFromBaseType(value);
+		}
+	}
+
+	function len()
+	{
+		return this.__Data.len();
+	}
+
+	function getDataArray()
+	{
+		return this.__DataArray;
+	}
```

Comment:
Why does this need to be public?

Reply:
It's used in `serde_emulator` to check the validity of the data during `__readData`